File: app/src/main/java/com/coinninja/coinkeeper/util/uuid/UUIDGenerator.java
package com.coinninja.coinkeeper.util.uuid;

import java.util.UUID;

import javax.inject.Inject;

public class UUIDGenerator {

    @Inject
    UUIDGenerator() {
    }

    public String generate() {
        return UUID.randomUUID().toString();
    }

}

File: app/src/main/java/com/coinninja/coinkeeper/util/uuid/UUIDGenerator.kt
package com.coinninja.coinkeeper.util.uuid

import app.dropbit.annotations.Mockable
import java.util.*
import javax.inject.Inject

@Mockable
class UUIDGenerator @Inject internal constructor() {
    fun generate(): String {
        return UUID.randomUUID().toString()
    }
}